p2panda-rs/src/schema/system.rs

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use crate::hash::Hash;
+use crate::schema::fields::Type;
+use crate::schema::SchemaBuilder;
+
+use super::Schema;
+
+/// Make CDDL definition for operations from a given schema
+fn make_schema_cddl(
+    schema_name: &str,
+    hash: &str,
+    fields_schema_builder: &SchemaBuilder,
+    can_update: bool,
+    can_delete: bool,
+) -> String {
+    let mut schema = format!(
+        r#"; System schema {name}
+
+        {name} = {{
+            schema: "{hash}",
+            version: 1,
+            operation-body-{name}-{hash},
+        }}
+
+        hash = tstr .regexp "[0-9a-f]{{68}}"
+
+        operation-body-{name}-{hash} = (
```

Comment:
I wasn't able to make this line 

```
operation-body-{name}-{hash} //= (
```

even though the spec says I should be able to do that :(

> It is not an error if a name is first used with a "/=" or "//=" (there is no need to "create it" with "=").
[Section 2.2.2](https://datatracker.ietf.org/doc/html/rfc8610#section-2.2.2)